coreax/kernels/scalar_valued.py

```diff
@@ -181,6 +182,142 @@ def divergence_x_grad_y_elementwise(self, x: ArrayLike, y: ArrayLike) -> Array:
+        ) / div**3
+        return first_term - second_term
+
+
+class MaternKernel(ScalarValuedKernel):
+    r"""
+    Define Matérn kernel with smoothness parameter a multiple of :math:`\frac{1}{2}`.
+
+    Given :math:`\lambda =``length_scale` and :math:`\rho =``output_scale`, the Matérn
+    kernel with smoothness parameter :math:`\nu` set to be a multiple of
+    :math:`\frac{1}{2}`, i.e. :math:`\nu = p + \frac{1}{2}` where
+    :math:`p`=``degree``:math:`\in\mathbb{N}`, is defined as
+    :math:`k: \mathbb{R}^d \times \mathbb{R}^d \to \mathbb{R}`,
+
+    .. math::
+        k(x, y) = \rho^2 * \exp\left((-\frac{\sqrt{2p+1}||x-y||}{\lambda}\right)
+        \frac{p!}{(2p)!}\sum_{i=0}^p\frac{(p+i)!}{i!(p-i)!}
+        \left(2\sqrt{2p+1}\frac{||x-y||}{\lambda}\right)^{p-i}
+
+    where :math:`||\cdot||` is the usual :math:`L_2`-norm.
+
+    :param length_scale: Kernel smoothing/bandwidth parameter, :math:`\lambda`, must be
+        positive
+    :param output_scale: Kernel normalisation constant, :math:`\rho`, must be positive
+    :param degree: Kernel degree, :math:`p`, must be a non-negative integer
+    """
+
+    length_scale: float = eqx.field(default=1.0, converter=float)
+    output_scale: float = eqx.field(default=1.0, converter=float)
+    degree: int = 1
+
+    def __check_init__(self):
+        """Check attributes are valid."""
+        if self.length_scale <= 0:
+            raise ValueError("'length_scale' must be positive")
+        if self.output_scale <= 0:
+            raise ValueError("'output_scale' must be positive")
+        if not isinstance(self.degree, int) or self.degree < 0:
+            raise ValueError("'degree' must be a non-negative integer")
+
+    def _compute_summation_term(self, body: float, iteration: ArrayLike) -> Array:
+        r"""
+        Compute the summation term of the Matérn kernel for a given iteration.
+
+        Given :math:`p`=``degree``:math:`\in\mathbb{N}`, compute
+
+        .. math::
+            \sum_{i=0}^p\frac{(p+i)!}{i!(p-i)!}
+            \left(2\sqrt{2p+1}\frac{||x-y||}{\lambda}\right)^{p-i}.
+
+        :param body: Float representing
+            :math:`\left(\sqrt{2p+1}\frac{||x-y||}{\lambda}\right)`
+        :param iteration: Current iteration
+        """
+        factorial_term = factorial(self.degree + iteration) / (
+            factorial(iteration) * factorial(self.degree - iteration)
+        )
+        distance_term = (2 * body) ** (self.degree - iteration)
+        return factorial_term * distance_term
+
+    @override
+    def compute_elementwise(self, x: ArrayLike, y: ArrayLike) -> Array:
+        norm = jnp.linalg.norm(jnp.subtract(x, y))
+        body = (jnp.sqrt(2 * self.degree + 1) * norm) / self.length_scale
+        factor = (
+            self.output_scale**2
+            * jnp.exp(-body)
+            * factorial(self.degree)
+            / factorial(2 * self.degree)
+        )
+
+        summation = 1.0
+        if self.degree > 0:
```

Comment:
No test hits the case where self.degree is not greater than 0

Reply:
This should occur in the mean and gradient tests as the sampled degree parameter is not zero

Reply:
I can add analytical tests as well?

Reply:
From a coverage report I don't see it being hit - I think an analytical test would probably be best if possible.